### ISSUE TITLE
web_connectivity: investigate mail.voila.fr failure

### ISSUE THREAD

**Issue by bassosimone** [measurement-kit/measurement-kit#1196]
This is the result JSON with MK v0.4.3. The basic problem seems to be `http_parser_error`.

```json
{
    "annotations": {
        "platform": "macos"
    },
    "data_format_version": "0.2.0",
    "input": "http://mail.voila.fr/",
    "measurement_start_time": "2017-04-19 16:57:51",
    "probe_asn": "AS0",
    "probe_cc": "ZZ",
    "probe_ip": "127.0.0.1",
    "report_id": "20170419T165751Z_AS0_69BOknGgvhfxOUCS0gI9X3QHAXC2xPjbQ1tQ7Rq0SmVJv4ULKF",
    "software_name": "measurement_kit",
    "software_version": "0.4.3",
    "test_keys": {
        "accessible": false,
        "agent": "agent",
        "blocking": "http-failure",
        "body_length_match": null,
        "client_resolver": "193.70.152.61",
        "control": {
            "dns": {
                "addrs": [
                    "193.252.22.171"
                ],
                "failure": null
            },
            "http_request": {
                "body_length": 37407,
                "failure": null,
                "headers": {
                    "Content-Type": "text/html",
                    "Date": "Wed, 19 Apr 2017 16:57:56 GMT",
                    "Expires ": "Wed, 19 Apr 2017 17:57:56 GMT",
                    "Server": "Apache",
                    "Status": "200",
                    "Vary": "Accept-Encoding"
                },
                "status_code": 200,
                "title": "Sortir avec Orange : le site idal pour chercher des ides de sorties en ville"
            },
            "tcp_connect": {
                "193.252.22.171:80": {
                    "failure": null,
                    "status": true
                }
            }
        },
        "control_failure": null,
        "dns_consistency": "consistent",
        "dns_experiment_failure": null,
        "headers_match": null,
        "http_experiment_failure": "http_parser_error",
        "queries": [
            {
                "answers": [
                    {
                        "answer_type": "A",
                        "ipv4": "193.252.22.171",
                        "ttl": 205
                    }
                ],
                "failure": null,
                "hostname": "mail.voila.fr",
                "query_type": "A",
                "resolver_hostname": "8.8.8.8",
                "resolver_port": 53
            }
        ],
        "requests": [
            {
                "failure": "http_parser_error"
            }
        ],
        "retries": null,
        "socksproxy": null,
        "status_code_match": null,
        "tcp_connect": [
            {
                "ip": "193.252.22.171",
                "port": 80,
                "status": {
                    "blocked": false,
                    "failure": null,
                    "success": true
                }
            }
        ],
        "title_match": null
    },
    "test_name": "web_connectivity",
    "test_runtime": 4.40001082420349,
    "test_start_time": "2017-04-19 16:57:49",
    "test_version": "0.0.1"
}
```

**Comment by bassosimone** [comment]
The buffer that causes the parser to fail is:

```
HTTP/1.1 200 OK
Date: Tue, 02 May 2017 14:34:47 GMT
Server: Apache
Expires : Tue, 02 May 2017 15:34:47 GMT
Status: 200
Vary: Accept-Encoding
Connection: close
Transfer-Encoding: chunked
Content-Type: text/html

9282
<!DOCTYPE html>
<html lang="fr">
[...]
```

The issue is at byte 77. Accounting for the `\r\n`, the error is cased by the space before the colon in the `Expires` header. I will see if it's possible to convince the parser to be more tolerant (I believe it's possible).

**Comment by bassosimone** [comment]
It should be fixed by #1225. Excerpt of running the test with same host and fix applied:

```JSON
{
    "annotations": {
        "engine_name": "libmeasurement_kit",
        "engine_version": "0.6.0",
        "engine_version_full": "v0.6.0-1-g5cf8599a",
        "platform": "macos"
    },
    "data_format_version": "0.2.0",
    "input": "http://mail.voila.fr/",
    "measurement_start_time": "2017-05-10 18:16:02",
    "probe_asn": "AS3269",
    "probe_cc": "IT",
    "probe_ip": "127.0.0.1",
    "report_id": "20170510T181602Z_AS3269_LHBlStmlKziJMxOxJkt09krTLfKiNufKe9CiH0rHez3ESHf
uxV",
    "software_name": "measurement_kit",
    "software_version": "0.6.0",
    "test_keys": {
        "accessible": true,
        "agent": "agent"
}}
```

**Comment by bassosimone** [comment]
Closing because #1225 has been merged
